src/main/java/coloringpage/ImageController.java

```diff
@@ -25,6 +25,7 @@ public BufferedImage getFinalImage() throws Exception {
         if (lineDrawing != null){
             return lineDrawing;
         } else {
+            JOptionPane.showMessageDialog(null, "Line drawing image is null.");
             throw new Exception("Line drawing image is null");
```

Comment:
If you get rid of the throw new Exception line, does the method complain that there's no return?

Reply:
yes - but we can also do this:
```
public BufferedImage getFinalImage() {
        if (lineDrawing == null) {
            JOptionPane.showMessageDialog(null, "Line drawing image is null.");
        }
        return lineDrawing;
    }

```

what do you think is better?

Reply:
Hm, not really sure....I'm leaning a little more into what you suggested. If the image is null and gets passed to the JLabel then I assume it just shows up blank? So long as it doesnt cause errors anything that's fine with me